include/ck_tile/core/numeric/vector_type.hpp

```diff
@@ -34,7 +35,11 @@ template <typename T_, index_t N_>
 struct ext_vector
 {
     static constexpr index_t N = N_;
-    using value_type           = typename native_t<remove_cvref_t<T_>>::type;
+    // struct type is not supported for ext_vector
+    using value_type =
+        std::conditional_t<std::is_same_v<typename native_t<remove_cvref_t<T_>>::type, pk_int4_t>,
+                           int8_t,
+                           typename native_t<remove_cvref_t<T_>>::type>;
```

Comment:
You could have specialization here for non POD types (like basic int, int16 etc) where you would do:
```suggestion
    // struct type is not supported for ext_vector
    using value_type = typename T_::type;
```

Reply:
Generally with conditional_t it is not possible because conditional_t evaluates the both of types (for true and false), If I use ::type then it will check for float::type/int::type etc. Which has no such member